PDBminer

```diff
@@ -98,25 +98,42 @@ def check_input_file(df):
     df = df.fillna("N/A")
 
     problems = []
+    hugo_name_list = []
     
-    for i in range(len(df)):
-        if type(df.hugo_name[i]) != str:
-            problems.append(f"there is a problem with the input on the line of {df.loc[i]['hugo_name']}")
+    for i in range(len(df)):    
         if type(df.uniprot[i]) != str:
-            problems.append(f"there is a problem with the uniprot input on the line of {df.loc[i]['hugo_name']}")
+            problems.append(f"there is a problem with the uniprot input on the line of {df.loc[i]['uniprot']}")
         
+        if ('hugo_name' in df.columns) == False:
```

Comment:
you can also use BioPython to parse these files, e.g.

```
from Bio import SwissProt
u = list(SwissProt.parse(open('P12345.txt')))
u[0].gene_name
```

I'm sure it's possible to do it from a text stream instead of saving it. This might be more robust. Another option is to download the file as a json which is also possible, e.g.:

```
response = requests.get('https://www.uniprot.org/uniprot/P12345.json')
d = json.loads(response.text)
print(d['genes'][0]['geneName']['value'])
```

Reply:
This has been updated with the .json file import.